### ISSUE TITLE
Code cells disappear

### ISSUE THREAD

**Issue by jmontoyam** [jupyterlab/jupyterlab#8331]
## Description

Hi,

first of all, thank you very much for putting the effort to maintain this amazing project!. Recently, I started to notice a weird behavior in JupyterLab: some of the code cells disappear. In the first image, you can see a very simple example where I create a JupyterLab notebook, I save it and then I close JupyterLab server and the JupyterLab client. 

![screen1](https://user-images.githubusercontent.com/3098532/80691750-f6ffa080-8ad0-11ea-9c59-16ee9f4cab72.png)

Two minutes later I open the recently created notebook, and as you can see in the second image, the code cell number 3 has disappeared:

![screen2](https://user-images.githubusercontent.com/3098532/80691827-11d21500-8ad1-11ea-9dbb-32c34bcde938.png)

But, if I open the same notebook using the classic Jupyter Notebook instead of JupyterLab, I can see that the code cell number 3 is still there, as you can see in the following image:

![screen3](https://user-images.githubusercontent.com/3098532/80692007-4645d100-8ad1-11ea-9198-cb345a622a4a.png)

Then, one more time I open the notebook using JupyterLab (the code cell number 3 is still missing), and I re-run all the cells. As you can see in the following image, according to JupyterLab there are only 3 code cells (it does not see at all the previous code cell number 3 showed in the first image):

![screen4](https://user-images.githubusercontent.com/3098532/80692487-ec91d680-8ad1-11ea-90a9-caf007e48f0e.png)

But If I open the notebook showed in the image above, using this time the classic Jupyter Notebook, once again the outputs are different, I can see in the notebook one extra code cell not shown in JupyterLab, this extra cell is the one that is missing in JupyterLab:

![screen5](https://user-images.githubusercontent.com/3098532/80692884-8a85a100-8ad2-11ea-9a66-3d756b6cc988.png)

## Reproduce

I described above how to reproduce the bug (please see the attached images).

## Expected behavior

All the code cells should remain in the notebook, none of them should be missing after opening a previous saved notebook, as it is shown in the classic Jupyter Notebook images attached above.

## Context

- Operating System and version: The jupyterlab server is running on a remote server with Ubuntu 18.04.4 LTS (I connect to the remote server using a VPN).

- Browser and version: I connect to the remote server from my local machine running windows 10, I use chrome version 81.0.4044.129 (Official Build) (64-bit)

- JupyterLab version:
jupyter lab --version
2.0.0

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.

jupyter troubleshoot
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(140): Could not remove or rename /opt/python/miniconda3/conda-meta/jupyterlab_server-1.1.1-py_0.json.  Please remove this file manually (you may need to reboot to free file handles)
$PATH:
	/opt/python/miniconda3/bin
	/opt/python/miniconda3/condabin
	/usr/local/cuda/bin
	/opt/nvidia/nsight-compute/2019.5.0
	/home/xxxxx/.local/bin
	/home/xxxxx/bin
	/usr/local/sbin
	/usr/local/bin
	/usr/sbin
	/usr/bin
	/sbin
	/bin
	/usr/games
	/usr/local/games
	/snap/bin

sys.path:
	/opt/python/miniconda3/bin
	/opt/python/miniconda3/lib/python37.zip
	/opt/python/miniconda3/lib/python3.7
	/opt/python/miniconda3/lib/python3.7/lib-dynload
	/home/xxxxx/.local/lib/python3.7/site-packages
	/opt/python/miniconda3/lib/python3.7/site-packages

sys.executable:
	/opt/python/miniconda3/bin/python

sys.version:
	3.7.6 (default, Jan  8 2020, 19:59:22) 
	[GCC 7.3.0]

platform.platform():
	Linux-4.15.0-55-generic-x86_64-with-debian-buster-sid

which -a jupyter:
	/opt/python/miniconda3/bin/jupyter

pip list:
	Package                Version            
	---------------------- -------------------
	asn1crypto             1.3.0              
	attrs                  19.3.0             
	backcall               0.1.0              
	bleach                 3.1.4              
	bokeh                  2.0.1              
	certifi                2020.4.5.1         
	cffi                   1.14.0             
	chardet                3.0.4              
	click                  7.1.1              
	cloudpickle            1.3.0              
	colorcet               2.0.2              
	conda                  4.8.3              
	conda-package-handling 1.6.0              
	cryptography           2.8                
	cycler                 0.10.0             
	cytoolz                0.10.1             
	dask                   2.15.0             
	datashader             0.10.0             
	datashape              0.5.4              
	decorator              4.4.2              
	defusedxml             0.6.0              
	distributed            2.15.0             
	entrypoints            0.3                
	fsspec                 0.7.1              
	graphviz               0.13.2             
	HeapDict               1.0.1              
	holoviews              1.13.2             
	hvplot                 0.5.2              
	idna                   2.8                
	imageio                2.8.0              
	importlib-metadata     1.6.0              
	ipykernel              5.2.1              
	ipython                7.13.0             
	ipython-genutils       0.2.0              
	ipywidgets             7.5.1              
	jedi                   0.17.0             
	Jinja2                 2.11.2             
	joblib                 0.14.1             
	json5                  0.9.0              
	jsonschema             3.2.0              
	jupyter-bokeh          2.0.1              
	jupyter-client         6.1.3              
	jupyter-core           4.6.3              
	jupyterlab             2.0.0              
	jupyterlab-server      1.0.7              
	kiwisolver             1.1.0              
	llvmlite               0.32.0             
	locket                 0.2.0              
	Markdown               3.1.1              
	MarkupSafe             1.1.1              
	matplotlib             3.2.1              
	mistune                0.8.4              
	msgpack                1.0.0              
	multipledispatch       0.6.0              
	nbconvert              5.6.1              
	nbformat               5.0.6              
	networkx               2.4                
	notebook               6.0.3              
	numba                  0.49.0             
	numpy                  1.18.1             
	olefile                0.46               
	opencv-python          4.2.0.34           
	packaging              20.3               
	pandas                 1.0.3              
	pandocfilters          1.4.2              
	panel                  0.9.5              
	param                  1.9.3              
	parso                  0.7.0              
	partd                  1.1.0              
	pexpect                4.8.0              
	pickleshare            0.7.5              
	Pillow                 7.0.0              
	pip                    20.0.2             
	prometheus-client      0.7.1              
	prompt-toolkit         3.0.5              
	psutil                 5.7.0              
	ptyprocess             0.6.0              
	pycosat                0.6.3              
	pycparser              2.19               
	pyct                   0.4.6              
	Pygments               2.6.1              
	pyOpenSSL              19.1.0             
	pyparsing              2.4.6              
	pyrsistent             0.16.0             
	PySocks                1.7.1              
	python-dateutil        2.8.1              
	pytz                   2019.3             
	pyviz-comms            0.7.4              
	PyWavelets             1.1.1              
	PyYAML                 5.3.1              
	pyzmq                  19.0.0             
	requests               2.22.0             
	ruamel-yaml            0.15.87            
	scikit-image           0.16.2             
	scikit-learn           0.22.1             
	scipy                  1.4.1              
	Send2Trash             1.5.0              
	setuptools             45.2.0.post20200210
	sidecar                0.4.0              
	six                    1.14.0             
	sortedcontainers       2.1.0              
	tblib                  1.6.0              
	terminado              0.8.3              
	testpath               0.4.4              
	toolz                  0.10.0             
	tornado                6.0.4              
	tqdm                   4.42.1             
	traitlets              4.3.3              
	typing-extensions      3.7.4.1            
	urllib3                1.25.8             
	wcwidth                0.1.9              
	webencodings           0.5.1              
	wheel                  0.34.2             
	widgetsnbextension     3.5.1              
	xarray                 0.15.1             
	zict                   2.0.0              
	zipp                   3.1.0

conda list:
	# packages in environment at /opt/python/miniconda3:
	#
	# Name                    Version                   Build  Channel
	_libgcc_mutex             0.1                        main  
	asn1crypto                1.3.0                    py37_0  
	attrs                     19.3.0                     py_0    conda-forge
	backcall                  0.1.0                      py_0    conda-forge
	blas                      1.0                    openblas  
	bleach                    3.1.4              pyh9f0ad1d_0    conda-forge
	bokeh                     2.0.1                    py37_0  
	ca-certificates           2020.1.1                      0  
	certifi                   2020.4.5.1               py37_0  
	cffi                      1.14.0           py37h2e261b9_0  
	chardet                   3.0.4                 py37_1003  
	click                     7.1.1                      py_0  
	cloudpickle               1.3.0                      py_0  
	colorcet                  2.0.2                      py_0  
	conda                     4.8.3                    py37_0  
	conda-package-handling    1.6.0            py37h7b6447c_0  
	cryptography              2.8              py37h1ba5d50_0  
	cycler                    0.10.0                   py37_0  
	cytoolz                   0.10.1           py37h7b6447c_0  
	dask                      2.15.0                     py_0  
	dask-core                 2.15.0                     py_0  
	datashader                0.10.0                     py_0  
	datashape                 0.5.4                    py37_1  
	dbus                      1.13.12              h746ee38_0  
	decorator                 4.4.2                      py_0    conda-forge
	defusedxml                0.6.0                      py_0    conda-forge
	distributed               2.15.0                   py37_0  
	entrypoints               0.3             py37hc8dfbb8_1001    conda-forge
	expat                     2.2.6                he6710b0_0  
	fontconfig                2.13.1            h86ecdb6_1001    conda-forge
	freetype                  2.9.1                h8a8886c_1  
	fsspec                    0.7.1                      py_0  
	glib                      2.63.1               h5a9c865_0  
	gst-plugins-base          1.14.5               h0935bb2_2    conda-forge
	gstreamer                 1.14.5               h36ae1b5_2    conda-forge
	heapdict                  1.0.1                      py_0  
	holoviews                 1.13.2                     py_0  
	hvplot                    0.5.2                      py_1  
	icu                       64.2                 he1b5a44_1    conda-forge
	idna                      2.8                      py37_0  
	imageio                   2.8.0                      py_0  
	importlib-metadata        1.6.0            py37hc8dfbb8_0    conda-forge
	importlib_metadata        1.6.0                         0    conda-forge
	ipykernel                 5.2.1            py37h43977f1_0    conda-forge
	ipython                   7.13.0           py37hc8dfbb8_2    conda-forge
	ipython_genutils          0.2.0                      py_1    conda-forge
	ipywidgets                7.5.1                      py_0  
	jedi                      0.17.0           py37hc8dfbb8_0    conda-forge
	jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
	joblib                    0.14.1                     py_0  
	jpeg                      9c                h14c3975_1001    conda-forge
	json5                     0.9.0                      py_0    conda-forge
	jsonschema                3.2.0            py37hc8dfbb8_1    conda-forge
	jupyter_bokeh             2.0.1                      py_0    bokeh
	jupyter_client            6.1.3                      py_0    conda-forge
	jupyter_core              4.6.3            py37hc8dfbb8_1    conda-forge
	jupyterlab                2.0.0                      py_1    conda-forge
	jupyterlab-server         1.0.7                    pypi_0    pypi
	kiwisolver                1.1.0            py37he6710b0_0  
	ld_impl_linux-64          2.33.1               h53a641e_7  
	libblas                   3.8.0               14_openblas    conda-forge
	libcblas                  3.8.0               14_openblas    conda-forge
	libedit                   3.1.20181209         hc058e9b_0  
	libffi                    3.2.1                hd88cf55_4  
	libgcc                    7.2.0                h69d50b8_2    conda-forge
	libgcc-ng                 9.1.0                hdf63c60_0  
	libgfortran-ng            7.3.0                hdf63c60_5    conda-forge
	liblapack                 3.8.0               14_openblas    conda-forge
	libopenblas               0.3.7                h5ec1e0e_6    conda-forge
	libpng                    1.6.37               hbc83047_0  
	libsodium                 1.0.17               h516909a_0    conda-forge
	libstdcxx-ng              9.1.0                hdf63c60_0  
	libtiff                   4.1.0                h2733197_0  
	libuuid                   2.32.1            h14c3975_1000    conda-forge
	libuv                     1.34.0               h516909a_0    conda-forge
	libxcb                    1.13                 h1bed415_1  
	libxml2                   2.9.9                hea5a465_1  
	llvmlite                  0.32.0           py37hd408876_0  
	locket                    0.2.0                    py37_1  
	markdown                  3.1.1                    py37_0  
	markupsafe                1.1.1            py37h8f50634_1    conda-forge
	matplotlib                3.2.1                         0    conda-forge
	matplotlib-base           3.2.1            py37h30547a4_0    conda-forge
	mistune                   0.8.4           py37h8f50634_1001    conda-forge
	msgpack-python            1.0.0            py37hfd86e86_1  
	multipledispatch          0.6.0                    py37_0  
	nbconvert                 5.6.1            py37hc8dfbb8_1    conda-forge
	nbformat                  5.0.6                      py_0    conda-forge
	ncurses                   6.2                  he6710b0_0  
	networkx                  2.4                        py_0  
	nodejs                    13.13.0              hf5d1a2b_0    conda-forge
	notebook                  6.0.3                    py37_0    conda-forge
	numba                     0.49.0           py37h0573a6f_0  
	numpy                     1.18.1           py37h8960a57_1    conda-forge
	olefile                   0.46                     py37_0  
	opencv-python             4.2.0.34                 pypi_0    pypi
	openssl                   1.1.1g               h7b6447c_0  
	packaging                 20.3                       py_0  
	pandas                    1.0.3            py37h0573a6f_0  
	pandoc                    2.9.2.1                       0    conda-forge
	pandocfilters             1.4.2                      py_1    conda-forge
	panel                     0.9.5                      py_1  
	param                     1.9.3                      py_0  
	parso                     0.7.0              pyh9f0ad1d_0    conda-forge
	partd                     1.1.0                      py_0  
	pcre                      8.43                 he6710b0_0  
	pexpect                   4.8.0            py37hc8dfbb8_1    conda-forge
	pickleshare               0.7.5           py37hc8dfbb8_1001    conda-forge
	pillow                    7.0.0            py37hb39fc2d_0  
	pip                       20.0.2                   py37_1  
	prometheus_client         0.7.1                      py_0    conda-forge
	prompt-toolkit            3.0.5                      py_0    conda-forge
	psutil                    5.7.0            py37h7b6447c_0  
	ptyprocess                0.6.0                   py_1001    conda-forge
	pycosat                   0.6.3            py37h7b6447c_0  
	pycparser                 2.19                     py37_0  
	pyct                      0.4.6                    py37_0  
	pygments                  2.6.1                      py_0    conda-forge
	pyopenssl                 19.1.0                   py37_0  
	pyparsing                 2.4.6                      py_0  
	pyqt                      5.9.2            py37h05f1152_2  
	pyrsistent                0.16.0           py37h8f50634_0    conda-forge
	pysocks                   1.7.1                    py37_0  
	python                    3.7.6                h0371630_2  
	python-dateutil           2.8.1                      py_0    conda-forge
	python_abi                3.7                     1_cp37m    conda-forge
	pytz                      2019.3                     py_0  
	pyviz_comms               0.7.4                      py_0  
	pywavelets                1.1.1            py37h7b6447c_0  
	pyyaml                    5.3.1            py37h7b6447c_0  
	pyzmq                     19.0.0           py37hac76be4_1    conda-forge
	qt                        5.9.7                h0c104cb_3    conda-forge
	readline                  7.0                  h7b6447c_5  
	requests                  2.22.0                   py37_1  
	ruamel_yaml               0.15.87          py37h7b6447c_0  
	scikit-image              0.16.2           py37h0573a6f_0  
	scikit-learn              0.22.1           py37h22eb022_0  
	scipy                     1.4.1            py37habc2bb6_0  
	send2trash                1.5.0                      py_0    conda-forge
	setuptools                45.2.0                   py37_0  
	sidecar                   0.4.0                    pypi_0    pypi
	sip                       4.19.8           py37hf484d3e_0  
	six                       1.14.0                   py37_0  
	sortedcontainers          2.1.0                    py37_0  
	sqlite                    3.31.1               h7b6447c_0  
	tbb                       2020.0               hfd86e86_0  
	tblib                     1.6.0                      py_0  
	terminado                 0.8.3            py37hc8dfbb8_1    conda-forge
	testpath                  0.4.4                      py_0    conda-forge
	tk                        8.6.10               hed695b0_0    conda-forge
	toolz                     0.10.0                     py_0  
	tornado                   6.0.4            py37h8f50634_1    conda-forge
	tqdm                      4.42.1                     py_0  
	traitlets                 4.3.3            py37hc8dfbb8_1    conda-forge
	typing_extensions         3.7.4.1                  py37_0  
	urllib3                   1.25.8                   py37_0  
	wcwidth                   0.1.9              pyh9f0ad1d_0    conda-forge
	webencodings              0.5.1                      py_1    conda-forge
	wheel                     0.34.2                   py37_0  
	widgetsnbextension        3.5.1                    py37_0  
	xarray                    0.15.1                     py_0  
	xraylib                   4.0.0            py37hf476568_0    conda-forge
	xz                        5.2.4                h14c3975_4  
	yaml                      0.1.7                had09818_2  
	zeromq                    4.3.2                he1b5a44_2    conda-forge
	zict                      2.0.0                      py_0  
	zipp                      3.1.0                      py_0    conda-forge
	zlib                      1.2.11               h7b6447c_3  
	zstd                      1.3.7                h0b5b093_0

</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.

(base) xxxxxxxxxx:~$ jupyter lab
[I 10:06:35.886 LabApp] JupyterLab extension loaded from /opt/python/miniconda3/lib/python3.7/site-packages/jupyterlab
[I 10:06:35.887 LabApp] JupyterLab application directory is /opt/python/miniconda3/share/jupyter/lab
[I 10:06:35.892 LabApp] Serving notebooks from local directory: /home/xxxxx
[I 10:06:35.892 LabApp] The Jupyter Notebook is running at:
[I 10:06:35.892 LabApp] http://xxxxxxxxxx:7820/
[I 10:06:35.892 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 10:09:03.704 LabApp] Creating new notebook in /
[I 10:09:03.968 LabApp] Kernel started: 7eee4252-2be4-42c9-a4f5-ebcaceaaf09c
[I 10:10:24.570 LabApp] Saving file at /reproducible_bug.ipynb
[I 10:11:42.645 LabApp] Starting buffering for 7eee4252-2be4-42c9-a4f5-ebcaceaaf09c:09eb86c3-7b03-4afb-88cb-92849c8009f9
[I 10:11:45.075 LabApp] Kernel shutdown: 7eee4252-2be4-42c9-a4f5-ebcaceaaf09c
[I 10:11:54.008 LabApp] Kernel started: 8e6f08c9-f132-4a1b-8cde-33a73cc21b53
[I 10:20:24.342 LabApp] Saving file at /reproducible_bug.ipynb
[I 10:21:18.069 LabApp] Starting buffering for 8e6f08c9-f132-4a1b-8cde-33a73cc21b53:7229b8c7-6dab-49db-ba2e-d37954a9a59d
[I 10:21:20.763 LabApp] Kernel shutdown: 8e6f08c9-f132-4a1b-8cde-33a73cc21b53
[I 10:21:26.441 LabApp] Kernel started: fbe4eed4-5337-49b7-b665-1cf0d585c898
[I 10:21:35.999 LabApp] Saving file at /reproducible_bug.ipynb
[I 10:24:36.506 LabApp] Starting buffering for fbe4eed4-5337-49b7-b665-1cf0d585c898:42c5f78e-0790-42ba-9330-abbfa3945d52
</pre>
</details>

<details><summary>Browser Output</summary>
<pre>
Paste the output from your browser Javascript console here.

The Javascript console is empty.

</pre>
</details>

Additional information:

(base) xxxxxxxxxx:~$ jupyter labextension list
JupyterLab v2.0.0
Known labextensions:
   app dir: /opt/python/miniconda3/share/jupyter/lab
        @bokeh/jupyter_bokeh v2.0.1  enabled  OK
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        @jupyter-widgets/jupyterlab-sidecar v0.5.0  enabled  OK
        @pyviz/jupyterlab_pyviz v1.0.1  enabled  OK

(base) xxxxxxxxxx:~$ jupyter serverextension list
config dir: /opt/python/miniconda3/etc/jupyter
    jupyterlab  enabled 
    - Validating...
      jupyterlab 2.0.0 OK

Thank you very much for your kind attention and your help.

**Comment by jasongrout** [comment]
Can you upload the notebook in question?

My guess is that bokeh is doing something to the HTML on the page that assumes the classic notebook, and it is affecting that cell in JupyterLab. Do you have a version of bokeh that supports JupyterLab 2.0?

**Comment by jasongrout** [comment]
Alternatively, can you reproduce this issue without using bokeh in the notebook?

Also, you could try going into the browser js console and looking for that missing cell. It would be interesting to see if that cell is in the browser page, but not showing, or if it is really gone.

**Comment by pybokeh** [comment]
Experiencing very similar problem here!  But, I am not using bokeh directly, but using Holoviz panel library which uses bokeh under the hood.  For me, it is usually the very first code input cell that gest hidden that usually contains my usual imports.  I have to use classic notebook to be able to see my hidden input cell.

**Comment by jmontoyam** [comment]
Thank you very much for your suggestions @jasongrout !. I have installed the right version of bokeh (2.0.1) compatible with JupyterLab 2 and its corresponding extension (@bokeh/jupyter_bokeh v2.0.1  enabled  OK). Taking into account @pybokeh comments, I made an example using only bokeh and JupyterLab behavies as expected (no code cells get loss). Then I made an example using only holoviews and I got the problem back (holoviews version 1.13.2 using the labextension @pyviz/jupyterlab_pyviz v1.0.1). I downgraded jupyterlab_pyviz extension to version 1.0.0 and rebuild JupyterLab and the problem was solved!. BTW @pybokeh, Holoviz panel also uses the jupyterlab_pyviz extension, so my solution could also solve your problem. I am sorry for the noise, at the end the issue was related with jupyterlab_pyviz extension.
